public/blog/announcing-full-portable-solidity-debugger.md

```diff
@@ -0,0 +1,79 @@
+# Announcing our Fully Featured, Portable Solidity Debugger
+
+Every one at Truffle is giddy right now. 
+
+----------------
+
+Since Truffle's inception, it's been our mission to build outstanding development tools for the Ethereum community. The analogy we always used was one of history: Ethereum development practices are years behind the rest of the industry, and it's our job to build tools that modernize our craft. Well today, Truffle's come one step closer to that goal. We're happy to announce the release of our fully featured, fully portable Solidity [debugger](/docs) and [debugging libraries](/docs). 
+
+    image or gif will live here
+
+### What We've Built
+
+If you've followed Truffle development, you're familiar with the `truffle debug` command. This was the first incarnation of a "debugger" built into Truffle, though it wasn't much more than a code tracer. Our new debugger comes with everything you expect :
+
+* Code stepping (over, into, out, next, instruction, etc.)
+* Current code location, including the address of the running contract
+* Breakpoints
+* Watch expressions
+* Variable inspection (stack, memory _and_ storage)
+* Custom expression evaluation using Solidity variables
+
+On top of that, our new debugger is fully portable, built to integrate with any Javascript project or editor, like VSCode. Lastly, we built it to work with **any Ethereum client**, so you could debug transactions against the main Ethereum network if you so desired. 
+
+### How it Works
+
+To debug your Solidity code, you first need to make a transaction. Ethereum is a world computer with a history of all execution, so we can pull all the information we need from past transactions. After obtaining the transaction hash, simply run the following to kick off your debugging experience:
+
+```shell
+$ truffle debug <tx hash>
+```
+
+This will open up the debugging console which you can use to your heart's content! 
+
+    Insert another picture.
+
+The above user interface gives you access to the debugger features you need to step through your contracts' code, set breakpoints, evaluate expressions and inspect Solidity variables, providing you a better sense of what your code is doing. Under the hood, our debugger is interacting with your Ethereum client to gather all the data necessary to provide this information to you in a suitable manner. For developers reading this, the debugging library stores all this data in Redux, making the data easy to manage and making the library super portable. 
```

Comment:
Singular possessive: "contract's"

Reply:
That'd be the code of one contract. `contracts'` would be the code of many contracts. I was shooting for the latter.